pkg/pool-stable-phantom/contracts/test/MockStablePhantomPool.sol

```diff
@@ -15,14 +15,57 @@
 pragma solidity ^0.7.0;
 pragma experimental ABIEncoderV2;
 
+import "@balancer-labs/v2-pool-utils/contracts/test/MockFailureModes.sol";
+
 import "../StablePhantomPool.sol";
 
-contract MockStablePhantomPool is StablePhantomPool {
+contract MockStablePhantomPool is StablePhantomPool, MockFailureModes {
     constructor(NewPoolParams memory params) StablePhantomPool(params) {
         // solhint-disable-previous-line no-empty-blocks
     }
 
-    function mockCacheTokenRateIfNecessary(IERC20 token) external {
+    function mockCacheTokenRateIfNecessary(IERC20 token) external whenNotInFailureMode(FailureMode.PRICE_RATE) {
         _cacheTokenRateIfNecessary(token);
```

Comment:
We should also add the modifier not to this external function but to the internal function that gets the rates and caches them. Otherwise, even if we set the price rate failure mode, the cache would not fail (which is not what would actually happen in production).

Reply:
Ok; also getTokenRate